manifests/init.pp

```diff
@@ -363,6 +372,8 @@
   Hash               $client_custom = {},
   Variant[Undef,Boolean] $client_deregister = undef,
   Variant[Undef,Hash] $client_deregistration = undef,
+  Variant[Undef,Boolean] $client_register = undef,
```

Comment:
What if you specify `client_register = true` and `client_deregister = true` ? 

Perhaps we need some logic so you can't do mutually exclusive things?

Reply:
Yes we can, but are they mutually exclusive?

Reply:
I'm not positive, what do the docs say?

Reply:
Docs don't mention any mutuality.
I think that registration and deregistration can both be configured, 
my concern is about the boolean register parameter, which is not mentioned in the docs (but deregister is).
My understanding based on https://sensuapp.org/docs/1.0/reference/clients.html#client-attributes is that we should just have registration , and not register, and they can cohexist with the de-reg* ones. But maybe it's better to ask to Sensu guys.

Reply:
@agoddard Could you please shed some light on this for us?

Reply:
@ghoneycutt @alvagante 

* `deregister` should be a boolean, it controls whether or not the client will send a "deregistration event" upon clean shutdown

* `register` should be a hash, the contents of which is passed along when the client first registers with the Sensu installation.

* `deregister` and `register` are not mutually exclusive